pkg/storage/chunk/client/azure/blob_storage_client.go

```diff
@@ -357,18 +351,25 @@ func (b *BlobStorage) getOAuthToken() (*azblob.TokenCredential, error) {
 	return &tc, nil
 }
 
-func (b *BlobStorage) fetchMSIToken() (*adal.ServicePrincipalToken, error) {
-	// msiEndpoint is the well known endpoint for getting MSI authentications tokens
-	// msiEndpoint := "http://169.254.169.254/metadata/identity/oauth2/token" for production Jobs
-	msiEndpoint, _ := adal.GetMSIVMEndpoint()
+func (b *BlobStorage) getServicePrincipalToken() (*adal.ServicePrincipalToken, error) {
+	var endpoint string
+	if b.cfg.Endpoint != "" {
+		endpoint = b.cfg.Endpoint
+	} else {
+		endpoint = defaultEndpoints[b.cfg.Environment]
+	}
+
+	resource := fmt.Sprintf("https://%s.%s", b.cfg.StorageAccountName, endpoint)
 
-	// both can be empty, systemAssignedMSI scenario
-	spt, err := adal.NewServicePrincipalTokenFromMSI(msiEndpoint, "https://storage.azure.com/")
-	if err != nil {
-		return nil, err
+	msiConfig := auth.MSIConfig{
+		Resource: resource,
+	}
+
+	if b.cfg.UserAssignedID != "" {
+		msiConfig.ClientID = b.cfg.UserAssignedID
 	}
```

Comment:
There's no need for the empty string check:

```suggestion
	msiConfig := auth.MSIConfig{
		Resource: resource,
		ClientID: b.cfg.UserAssignedID,
	}
```

Reply:
I agree that it's not actually needed, but I took the code from Thanos and appreciated that it made the flow easier to understand. I'm happy to remove it but just wanted to raise the readability question.